cmake/templates/OpenCVConfig-version.cmake.in

```diff
@@ -8,7 +8,3 @@ if(PACKAGE_FIND_VERSION VERSION_EQUAL PACKAGE_VERSION)
   set(PACKAGE_VERSION_EXACT True)
   set(PACKAGE_VERSION_COMPATIBLE True)
 endif()
-
-if(PACKAGE_FIND_VERSION VERSION_LESS PACKAGE_VERSION)
```

Comment:
In this case minor versions will be considered incompatible too (3.0 and 3.1).

Probably the expression could be transformed into something like this:
```.cmake
if (PACKAGE_FIND_VERSION_MAJOR EQUAL @OPENCV_VERSION_MAJOR@ 
    AND PACKAGE_FIND_VERSION VERSION_LESS PACKAGE_VERSION)
```